scapy/dadict.py

```diff
@@ -55,64 +73,84 @@ class DADict(object):
         ETHER_TYPES.IPv4 -> 2048
     """
     def __init__(self, _name="DADict", **kargs):
+        # type: (str, **Any) -> None
         self._name = _name
+        self.d = {}  # type: Dict[_K, _V]
         self.update(kargs)
 
     def ident(self, v):
+        # type: (_V) -> str
         """
         Return value that is used as key for the direct access
         """
-        return fixname(v)
+        if isinstance(v, (str, bytes)):
+            return fixname(v)
+        return "unknown"
 
     def update(self, *args, **kwargs):
+        # type: (*Dict[Any, Any], **Any) -> None
```

Comment:
Could we use _K and _V here instead of Any?

Reply:
Actually we can only use `_V`.
You can do `update(a=1)` -> `kwargs: Dict[str, _V]`
but not `update(1=2)`
so this function only works when `_K=str`...